src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -507,6 +508,12 @@ private bool MatchesSubtypeSuffix(MediaType set)
             return set.SubTypeSuffix.Equals(SubTypeSuffix, StringComparison.OrdinalIgnoreCase);
         }
 
+        private bool MatchesEitherSubtypeOrSuffix(MediaType set)
+        {
+            return set.SubTypeSuffix.Equals(SubType, StringComparison.OrdinalIgnoreCase) ||
+                set.SubTypeSuffix.Equals(SubTypeSuffix, StringComparison.OrdinalIgnoreCase);
```

Comment:
Looks backwards. How does this work when this method is called i.e. `! set.SubTypeSuffix.HasValue)`?

Reply:
Yeah you are correct.